### PR TITLE
docs: Restructure and update the documentation - 2023 edition

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR rationalizes the documentation a bit. The main points are

- Remove the documentation of the old standalone examples executables
- Restructure the core documentation to be less sprawling
- Move the figures from a central folder closer to the relevant source files
- Remove the full auto API documentation in favor of a combination of hard-coded documented symbols + an auto detection mechanism to catch when we references symbols somewhere
  - This reduces the time it takes to build the docs drastically
- Enable nitpicky generation, which will warn (and fail) if symbols are references that don't have a target.

- [x] Try to auto-generate API listings only for the elements we explicitly call out by class / func roles.

Blocked by:
- #2616 
- #2617 
- #2624

### CODE REVIEW

File: docs/requirements.in [hunk 1]
```diff
@@ -2,4 +2,4 @@ breathe
 myst-parser
 sphinx
 sphinx_rtd_theme
-docutils<0.17.0 # see https://github.com/readthedocs/sphinx_rtd_theme/issues/1115
+docutils
```
Comment (AJPfleger) [inline docs/requirements.in:5]: Is the issue not relevant for us anymore or should we maybe force `>0.17.0` just to be safe?

File: docs/core/eventdata.md
Comment (AJPfleger) [inline docs/core/eventdata.md:1]: Should we maybe add todos to this file?
```
:::{todo}
Write ...
:::
```

File: docs/examples/howto/howto.md [hunk 2]
```diff
@@ -1,24 +1,10 @@
...
-
-(howto-examples)=
-# How-to guides
-
-:::{attention}
-This section presents various workflows using the example executables.
-These executables are **deprecated**, and the python bindings should be used
-to run example workflows (see [](python-bindings)).
-:::
+# How-to guides for the Examples
```
Comment (AJPfleger) [inline docs/examples/howto/howto.md:1]: I think we are now not mentioning anywhere, that the examples are deprecated.

File: docs/core/material.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+(material-core)=
+
+# Material
+
+## Concept of passive material in tracking
+
+### Projective approximation of passive material
+% Link to instructions how to run the material mapping in the examples
```
Comment (AJPfleger) [inline docs/core/material.md:8]:
```suggestion
:::{todo}
Link to instructions how to run the material mapping in the examples
:::
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]
Maybe you could mention in the description, that you removed some of the examples.

File: docs/requirements.in (hunk 1 above)
Comment (paulgessinger) [reply docs/requirements.in:5]: Yeah let me do that.

File: docs/examples/howto/howto.md (hunk 2 above)
Comment (paulgessinger) [reply docs/examples/howto/howto.md:1]: The examples standalone executables are deprecated. Ultimately, I want to have updated guides like this for the python examples workflows.

**Review by AJPfleger — DISMISSED** [review]

**Comment by paulgessinger** [comment]
@AJPfleger can you click the button again? I've updated this after #2591 was merged.

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [922e6d1b56c9c0c9a01febac9e2d91e4777c7571]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33741171) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740428) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740427) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740426) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740425) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740424) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740423) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740422) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740421) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740420) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740419) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740418) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740417) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740416) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740415) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740414) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740413) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33740412) |

**Pull request merged**